ui/pages/confirmations/components/confirm/info/hooks/useDecodedTransactionData.ts

```diff
@@ -10,18 +10,23 @@ import { DecodedTransactionDataResponse } from '../../../../../../../shared/type
 import { useConfirmContext } from '../../../../context/confirm';
 import { hasTransactionData } from '../../../../../../../shared/modules/transaction.utils';
 
-export function useDecodedTransactionData(): AsyncResult<
-  DecodedTransactionDataResponse | undefined
-> {
+export function useDecodedTransactionData(
+  transactionType?: string,
```

Comment:
To make this more explicit and readable, could we use an object argument call this something like `transactionTypeFilter`?

We could also omit the filter completely and always decode but I appreciate this is a nice optimisation.

If we end up with many of usages of this hook, we may want to implement some kind of caching in the underlying background utils, or even move this to the `TransactionController`.